src/backend/distributed/citus--8.4-1--8.4-2.sql

```diff
@@ -0,0 +1,13 @@
+/* citus--8.3-1--8.4-1 */
+
+ALTER TABLE pg_catalog.pg_dist_colocation ADD distributioncolumncollation oid;
+UPDATE pg_catalog.pg_dist_colocation dc SET distributioncolumncollation = t.typcollation
```

Comment:
If we were ignoring collation before, this will break upgraded clusters since we are not resharding tables when upgrading. Can you look into this? Maybe also add a comment here explaining what we are doing here is the correct decision for upgraded tables.

Reply:
Does the collation affect hash distribution as well?

Reply:
Users should not've distributed non-default collated columns before this change, as until only recently we weren't even setting up workers to have collations on columns. It may make sense to have a check which errors upgrade if the user has a table's distribution column declared with a non default collation, just in case